sympy/matrices/matrices.py

```diff
@@ -1547,6 +1547,13 @@ def _matrix_pow_by_jordan_blocks(self, num):
         def jordan_cell_power(jc, n):
             N = jc.shape[0]
             l = jc[0, 0]
+            if (num < 0) == True and l == 0:
+                raise ValueError("Matrix det == 0; not invertible")
+            if (num < 2) == True:
```

Comment:
But there are other matrices where 2 does not suffice, namely nilpotent Jordan blocks of size > 2.